raft/leaderElection/leaderelection_test.go

```diff
@@ -0,0 +1,84 @@
+package leaderelection
+
+import (
+	"github.com/SUMUKHA-PK/Raft-Distributed-Consensus/types"
+	"math/rand"
+	"testing"
+	"time"
+)
+
+func TestIncrementTermAndBecomeCandidate(t *testing.T) {
+	config, _ := types.BuildConfigurationFromConfigFile("../../server.config.json")
+	serverData := types.BuildServerData(config)
+
+	host := "127.0.0.1:55551"
```

Comment:
@apoorva-mk 
Is hardcoding IP an issue here? We might have to consider building types.Configuration using test tables

Reply:
Agreed. Try taking it from the config